sympy/assumptions/handlers/matrices.py

```diff
@@ -38,6 +38,8 @@ def MatMul(expr, assumptions):
         factor, mmul = expr.as_coeff_mmul()
         if all(ask(Q.symmetric(arg), assumptions) for arg in mmul.args):
             return True
+        if ask(Q.diagonal(expr), assumptions):
```

Comment:
Deduction fails in the file satask.py. There are two dictionaries `can_be_true` and `can_be_false`, both are nonempty:

```
can_be_true = {
 ...
 Q.symmetric(V1.T*(V1 + V2)): True,
 Q.diagonal(V1.T*(V1 + V2)): False,
 ...
}

can_be_false = {
 ...
 Q.diagonal(V1.T*(V1 + V2)): False,
 Q.symmetric(V1.T*(V1 + V2)): False,
 ...
}
```
Then function returns `None`:
```
    if can_be_true and can_be_false:
        return None
```